common/src/main/java/com/microsoft/identity/common/internal/ui/browser/BrowserSelector.java

```diff
@@ -48,21 +48,31 @@
      * Searches through all browsers for the best match.
      * Browsers are evaluated in the order returned by the package manager,
      * which should indirectly match the user's preferences.
-     * First browser in the list will be preferred no matter weather or not the custom tabs supported.
+     * First matched browser in the list will be preferred no matter weather or not the custom tabs supported.
      *
      * @param context {@link Context} to use for accessing {@link PackageManager}.
      * @return Browser selected to use.
      */
-    public static Browser select(final Context context) throws ClientException {
+    public static Browser select(final Context context, final List<BrowserDescriptor> browserSafeList) throws ClientException {
         final List<Browser> allBrowsers = getAllBrowsers(context);
-        if (!allBrowsers.isEmpty()) {
-            Logger.verbose(TAG, "Select the browser to launch.");
-            Logger.verbosePII(TAG, "Browser's package name: " + allBrowsers.get(0).getPackageName() + " version: " + allBrowsers.get(0).getVersion());
-            return allBrowsers.get(0);
-        } else {
-            Logger.error(TAG, "No available browser installed on the device.", null);
-            throw new ClientException(ErrorStrings.NO_AVAILABLE_BROWSER_FOUND, "No available browser installed on the device.");
+
+        for (Browser browser : allBrowsers) {
+            for (BrowserDescriptor browserDescriptor : browserSafeList) {
+                if (browserDescriptor.matches(browser)) {
+                    Logger.verbose(TAG, "Select the browser to launch.");
+                    Logger.verbosePII(
```

Comment:
I think we can probably log this as non-PII without issue - is there something I'm not considering?